client/src/app/pages/applications/components/application-detail-drawer/components/download-button.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from "react";
+import { APPLICATIONS } from "@app/api/rest";
+import { Button } from "@patternfly/react-core";
+import spacing from "@patternfly/react-styles/css/utilities/Spacing/spacing";
+import { Application } from "@app/api/models";
+import { Spinner } from "@patternfly/react-core";
+
+export enum MimeType {
+  TAR = "tar",
+  YAML = "yaml",
+}
+function DownloadButton({
+  application,
+  mimeType,
+}: {
+  application: Application;
+  mimeType: MimeType;
+}) {
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleDownload = async () => {
+    let acceptHeader = "application/x-tar";
+
+    switch (mimeType) {
+      case MimeType.YAML:
+        acceptHeader = "application/x-yaml";
+        break;
+      case MimeType.TAR:
+      default:
+        acceptHeader = "application/x-tar";
+    }
+    setIsLoading(true);
+
+    try {
+      const response = await fetch(
+        `${APPLICATIONS}/${application?.id}/analysis/report`,
+        {
+          headers: {
+            Accept: acceptHeader,
+          },
+        }
+      );
+
+      if (!response.ok) {
+        throw new Error("Network response was not ok when downloading file.");
+      }
+
+      const blob = await response.blob();
+      const url = window.URL.createObjectURL(blob);
+      const a = document.createElement("a");
+      a.style.display = "none";
+      a.href = url;
+      a.download = `analysis-${application.name}.` + mimeType;
+
+      document.body.appendChild(a);
+      a.click();
```

Comment:
Oh wow, this is very clever but I'm surprised it's necessary. I think we've used the file-saver library in the past to achieve the same thing, but tbh this might be more elegant.

I wonder if you even need to do the fetch in JS... could we just have the href of this anchor element be the URL we're fetching and let the click event do all the work? If that works that would mean the browser handles the whole download and can show progress natively instead of having it appear as a delayed instant-download.